document/JS.bs

```diff
@@ -0,0 +1,935 @@
+                 1. For each |arg| in |arguments|,
+                     1. Append [=ToJSValue=](|arg|) to |jsArguments|.
+                 1. Let |ret| be ? [=Call=](|v|, undefined, |jsArguments|). If an exception is thrown, trigger a WebAssembly trap, and propagate the exception to the enclosing JavaScript.
+                 1. Let [|parameters|] → [|results|] be |functype|.
+                 1. If |results| is empty, return undefined.
+                 1. Otherwise, return [=ToWebAssemblyValue=](|ret|, |results|[0]).
+             1. Let |store| be the current agent cluster's [=associated store=].
+             1. Let (|store|, |funcaddr|) be alloc_func(|store|, |functype|, |hostfunc|).
+             1. Set the current agent cluster's [=associated store=] to |store|.
+         1. Let |externfunc| be the [=external value=] 𝖿𝗎𝗇𝖼 |funcaddr|
+         1. Append |externfunc| to |imports|.
+      1. If |externtype| is of the form 𝗀𝗅𝗈𝖻𝖺𝗅 |globaltype|,
+         1. If |globaltype| is [=i64=] or [=Type=](|v|) is not [=Number=], throw a {{LinkError}} exception.
+         1. Let |value| be [=ToWebAssemblyValue=](|v|, |globaltype|.valtype)
+         1. Assert: |globaltype|.mut is const, as verified by WebAssembly validation.
+         1. Let |store| be the current agent cluster's [=associated store=].
+         1. Let (|store|, |globaladdr|) be [=alloc_global=](|store|, |globaltype|, |value|).
+         1. Set the current agent cluster's [=associated store=] to |store|.
+         1. Let |externglobal| be 𝗀𝗅𝗈𝖻𝖺𝗅 |globaladdr|.
+         1. Append |externglobal| to |imports|.
+      1. If |externtype| is of the form 𝗆𝖾𝗆  |memtype|,
+         1. If |v| is not a {{Memory}} object, throw a {{LinkError}} exception.
+
+         Note: [=instantiate_module=] below will check the imported {{Memory}}'s size against the importing module's requirements.
+
+         2. Append |v| to |memories|.
+         1. Let |externmem| be the [=external value=] 𝗆𝖾𝗆  |v|.\[[Memory]].
+         1. Append |externmem| to |imports|.
+      1. Otherwise, |externtype| is of the form 𝗍𝖺𝖻𝗅𝖾 |tabletype|,
+         1. If |v| is not a {{Table}} instance, throw a {{LinkError}} exception.
+
+         Note: The table's length, etc is checked by [=instantiate_module=] below.
+
+         2. Append |v| to |tables|.
+         1. Let |tableaddr| be |v|.\[[Table]]
+         1. Let |externtable| be the [=external value=] 𝗍𝖺𝖻𝗅𝖾 |tableaddr|.
+         1. Append |externtable| to |imports|.
+         1. For each element |func| of |v|.\[[Values]],
+            1. If |func| is not null, append |func| to |funcs|.
+    1. Let (|store|, |instance|) be [=instantiate_module=](|store|, |module|, |imports|), up through step 16 of [=the instantiation algorithm=], but omitting execution of the start function.
+    1. If |instance| is [=error=], throw a {{LinkError}} exception, or other <a href="#errors">appropriate exception type</a>.
+    1. For each |tableaddr| in |instance|.𝗍𝖺𝖻𝗅𝖾𝖺𝖽𝖽𝗋𝗌,
```

Comment:
Wow, fancy Unicode.

Indenting in the output seems off, not sure why.

Reply:
Fixed, seems like Bikeshed really likes four-space indents.